### DISCUSSION TITLE
TRV devices become unresponsive when connected to coordinator

### DISCUSSION THREAD

**Discussion by moster11** [Koenkk/zigbee2mqtt#13414] [General]
Hey there,

I have a setup with several Popp-TRVs (identical to Danfoss Ally TRVs), lots of Aqara sensors and two routers, all connected to a Sonoff USB Dongle 3.0 Plus. I run Home Assistant and have automations that send temperatures measured by the Aqara sensors to the external_measured_room_sensor endpoint of the TRVs. 

Everything runs smoothly for some time - until the TRVs stop reacting to any command sent to them. They still report their states without any issue, it's just that I cannot send any command to them any more. Most notably this ONLY happens to TRVs which are connected directly to the coordinator, never if connected to a router. 

Looking at the devices tab shows recent "Last seen" values. I highlighted "Thermostat Badezimmer Fenster" as that's the one that's unresponsive (did an initial re-pairing of all devices this morning):

![image](https://user-images.githubusercontent.com/65911580/183062004-cf64e4a1-6a94-4db8-b79c-1f92ad40371f.png)

Looking at the map gives me a clear sign that this TRV will not react to my commands anymore. LQI is shown as 0 here, while the previous picture showed LQI 72:

![image](https://user-images.githubusercontent.com/65911580/183062402-7ac67ea2-0fec-446c-ad33-8cff9e321cb2.png)

I tried Z2M versions 1.27.0 and 1.26.0. The Sonoff dongle was on ZStack 20220219, now upgraded to dev branch 20220726. I resetted and repaired all devices numerous times, even changed to position of the coordinator to the other side of the house. Nothing helps, it's always the devices connected to the coordinator that stop reacting after some hours, at max one or two days. Only way to make them react again for some time is to take the batteries in and out.

Additionally I tried to set QoS for the devices to 1, set the MQTT version to 5 which also does not change their behaviour.

Anybody got a clue what's going on here?

**Comment by OZ1SEJ** [comment]
Can't contribute anything to a solution, but I have the exact same problem with Danfoss Ally on ZHA. Did you find a solution?

**Reply by moster11** [reply]
Please do yourself a big favor and don't use ZHA with these TRVs. Zigbee2MQTT will give you so many more attributes of the TRVs to configure and view the state of the TRVs.

Just look at [how many exposed entities you will get](https://www.zigbee2mqtt.io/devices/701721.html). I started with ZHA just like you and it was a total pain.

**Reply by OZ1SEJ** [reply]
Yes, thank you, I'd love to, but last time I tried, I found out that Zigbee2MQTT isn't plug-n-play, there's lots of configuration to be done, I'm not very well versed in MQTT, and I couldn't even make it start. But once I have more time, this i definitely on my list!

My comment was primarily to point out that I have the exact same problem in ZHA: When Danfoss Ally TRVs are connected to a Sonoff 3 dongle through a router, they work fine. But when connected directly to the coordinator, they keep sending status messages, but become uncommandable after a while. So I don't think this is a problem with Zigbee2MQTT, since this also happens with ZHA. Suggesting that the origin of this problem may be in a lower layer.

**Comment by antst** [comment]
actually, while it is show direct on map, story is a bit different.
they "die" when they try to switch from coordinator to router. At leas this is why I see on sniffing. then messages are not passing anymore, and, most likely, this is why it is seen as directly connected to coordinator with LQI of 0. Coordinator is not aware that now TRV sends through the router, because those messages never pass.

I struggle with the same issue (I have 15 of those), but only for 4 particular spots in the house where they jump to routers. I tried to remove routers (remove devices which are acting as routers and look attractive to TRV) and then things works.
But once I put router device back, after some time TRV tries to go via router and viola.

**Comment by moster11** [comment]
I found a solution for my use case. I had Zigbee2MQTT installed in Home Assistant OS which runs on a VM on my Synology NAS (DSM 7).

No matter if I used a Conbee II or the Sonoff Dongle I encountered the bespoken dropouts. As soon as I installed Zigbee2MQTT as a Docker image and set up my TRVs from there everything suddenly started to run without any flaws. Rock solid setup for more than 4 weeks already.

Seems to me that the USB passthrough to the VM was the root cause.
